cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -17,53 +15,234 @@
+                Bom bom = readJSONfile();
+                bom = addExternalReference(bom, url, hashes, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+
+            case "addComExtRef": {                                 //This adds extRef to Comp --> extRef
+                Bom bom = readJSONfile();
+                bom = addComExternalReference(bom, url, hashes, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+            default: {
+                System.out.println("Please enter a command.");
+            }
+        }
     }
 
-    static Bom createTestBom() {
+    static Bom createBom(String name, String version) {        //Create SBOM, test.JSON file
         Bom bom = new Bom();
+        System.out.println(bom.getBomFormat());
+        Component comp = new Component();
+        comp.setName(name);
+        comp.setVersion(version);
+        comp.setType(Component.Type.APPLICATION);
+        comp.setGroup("Eclipse Temurin");
+        comp.setAuthor("Vendor: Adoptium");
+        bom.addComponent(comp);
+        return bom;
+    }
+    static Bom addMetadata(Bom bom, String name) {               //Method to store metadata -->  name
+        comp.setName(name);
+        comp.setType(Component.Type.APPLICATION);
+        OrganizationalEntity org = new OrganizationalEntity();
+        org.setName("Eclipse Foundation");
+        org.setUrls(Collections.singletonList("https://www.eclipse.org/"));
+        meta.setManufacture(org);
+        meta.setComponent(comp);
+        bom.setMetadata(meta);
+        return bom;
+    }
+    static Bom addMetadataProperty(Bom bom, String name, String value) {     //Method to store metadata --> Properties List --> name-values
+        meta = bom.getMetadata();
+        prop1.setName(name);
+        prop1.setValue(value);
+        meta.addProperty(prop1);
+        bom.setMetadata(meta);
+        return bom;
+    }
 
-        Component comp1 = new Component();
-        comp1.setName("TestComponent");
-        comp1.setVersion("1.0.0");
-        comp1.setType(Component.Type.APPLICATION);
-        comp1.setAuthor("Adoptium");
-
-        bom.addComponent(comp1);
+    static Bom addComponent (Bom bom, String name) {                    //Method to store Component ->name
+        comp.setType(Component.Type.APPLICATION);
+        comp.setName(name);
+        bom.addComponent(comp);
+        return bom;
+    }
+    static Bom addComponentProperty(Bom bom, String name, String value) {     //Method to store Component->Property->name-value pairs
+        prop1.setName(name);
+        prop1.setValue(value);
+        prop.add(prop1);
+        comp.setProperties(prop);
+        bom.addProperty(prop1);
```

Comment:
This property is on the top level BOM, not the Component? need to identify which component?